src/gui/src/layoutViewer.cpp

```diff
@@ -1983,38 +1983,48 @@ void LayoutViewer::drawBlock(QPainter* painter,
   }
 }
 
-void LayoutViewer::drawAccessPoints(Painter& painter)
+void LayoutViewer::drawAccessPoints(Painter& painter, const std::vector<odb::dbInst*>& insts)
 {
   const int shape_limit = shapeSizeLimit();
-  const int shape_size = 100;
-  if (shape_limit > shape_size)
+  const int shape_size = 100; // units DBU
+  if (shape_limit > shape_size) {
     return;
-  for (auto term : block_->getITerms()) {
-    for (auto ap : term->getPrefAccessPoints()) {
-      if (options_->isVisible(ap->getLayer())) {
-        auto color = ap->hasAccess() ? gui::Painter::green : gui::Painter::red;
-        painter.setPen(color, /* cosmetic */ true);
-        Point pt = ap->getPoint();
-        odb::dbTransform xform;
-        int x, y;
-        term->getInst()->getLocation(x, y);
-        xform.setOffset({x, y});
-        xform.setOrient(odb::dbOrientType(odb::dbOrientType::R0));
-        xform.apply(pt);
-        painter.drawX(pt.x(), pt.y(), shape_size);
+  }
+
+  const Painter::Color has_access = Painter::green;
+  const Painter::Color not_access = Painter::red;
+
+  auto draw = [&](odb::dbAccessPoint* ap, const odb::dbTransform& transform) {
+    if (ap == nullptr) {
+      return;
+    }
+    if (!options_->isVisible(ap->getLayer())) {
+      return;
+    }
+
+    Point pt = ap->getPoint();
+    transform.apply(pt);
+
+    auto color = ap->hasAccess() ? has_access : not_access;
```

Comment:
Do we ever have APs in the db with no access?

Reply:
@arlpetergadfort @osamahammad21 any response?

Reply:
@maliberty the original code had this, so I copied it forward, but it would be up to @osamahammad21 as to whether this actually occurs.

Reply:
@osamahammad21 please respond to "whether this actually occurs"